python_modules/dagster/dagster/_core/definitions/declarative_automation/operands/operands.py

```diff
@@ -193,6 +194,21 @@ async def compute_subset(self, context: AutomationContext) -> EntitySubset:  # p
         )
 
 
+@whitelist_for_serdes
+@record
+class FreshnessStateCondition(SubsetAutomationCondition):
```

Comment:
For typing stuff, this should be a `SubsetAutomationCondition[AssetKey]` to indicate that this can only be applied to assets.

Also I think the condition this is most similar to conceptually is the `CheckResultCondition`, so it might be best to name it consistently with that (FreshnessResultCondition)